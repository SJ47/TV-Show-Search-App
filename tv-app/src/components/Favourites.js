import React from 'react'

// This component will render the view all the users favourite shows as clickable image thumbnails

const Favourites = ({ favouriteShows, onFavImageClicked }) => {

    const favouriteShowsImages = favouriteShows.map((favouriteShow) => {
        return <img onClick={() => { onFavImageClicked(favouriteShow) }} className="fav-image" src={favouriteShow.show.image.medium} key={favouriteShow.show.id} alt="" height="100px"></img>
    })

    // Render view of a users favourite shows
    return (
        <>
            <h2 className="favouriteShowsTitle">Favourite Shows</h2>
            <hr />
            <div className="favContainer">
                {favouriteShowsImages}
            </div>
        </>
    )
}

export default Favourites
