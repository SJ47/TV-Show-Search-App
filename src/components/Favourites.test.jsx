import { render } from '@testing-library/react';
import Favourites from './Favourites';

const mockOnFavImageClicked = jest.fn();
const favouriteShows = []

describe('Favourites', () => {
  it('should render styles correctly', () => {
    const { container } = render(
        <Favourites favouriteShows={favouriteShows} onFavImageClicked={mockOnFavImageClicked}/>
    );
    expect(container).toMatchSnapshot();
  });
});
