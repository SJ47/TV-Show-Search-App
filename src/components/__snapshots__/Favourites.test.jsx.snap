// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites should render styles correctly 1`] = `
<div>
  <h2
    class="favouriteShowsTitle"
  >
    Favourite Shows
  </h2>
  <hr />
  <div
    class="favContainer"
  />
</div>
`;
